### PR TITLE
Cluster creation in an environment

### PR DESCRIPTION
**Pull request by schfeca75**
@attilapalfi92 pls review

### CODE REVIEW

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/Stack.java [hunk 1]
```diff
@@ -181,6 +182,9 @@
     @JoinColumn(name = "createdBy")
     private User creator;
 
+    @ManyToOne
+    private EnvironmentView environment;
```
Comment (attilapalfi92) [inline core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/Stack.java]: No `@JoinColumn` is needed here?

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/StackSensitiveDataPropagator.java [hunk 2]
```diff
@@ -16,8 +18,9 @@
     @Inject
     private CredentialSourceDecorator credentialSourceDecorator;
 
-    public Stack propagate(CredentialSourceRequest credentialSourceRequest, Stack stack, Workspace workspace) {
-        if (credentialSourceRequest != null) {
+    public Stack propagate(StackRequest stackRequest, Stack stack, Workspace workspace) {
+        CredentialSourceRequest credentialSourceRequest = stackRequest.getCredentialSource();
+        if (StringUtils.isEmpty(stackRequest.getEnvironment()) && credentialSourceRequest != null) {
```
Comment (attilapalfi92) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/StackSensitiveDataPropagator.java]: Why is the additional condition needed here? Decorator does not use the environment if I am not terribly mistaken.

File: core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/Stack.java (hunk 1 above)
Comment (attilapalfi92) [reply core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/Stack.java]: It finds out from the variable name `environment`

**Comment by attilapalfi92** [comment]
What's left?

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/StackSensitiveDataPropagator.java (hunk 2 above)
Comment (schfeca75) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/StackSensitiveDataPropagator.java]: This code / class will be removed during API clarification (it is from the HDC era), but till that if environment is given for a stack, then the environment's credential will be used for the stack.

**Comment by schfeca75** [comment]
Nothing!

**Comment by attilapalfi92** [comment]
LGTM

**Pull request merged**